Weaver/RealmWeaver.Fody/ModuleWeaver.cs

```diff
@@ -965,18 +994,39 @@ private TypeDefinition WeaveRealmObjectHelper(TypeDefinition realmObjectType, Me
             il.Append(il.Create(OpCodes.Castclass, ModuleDefinition.ImportReference(realmObjectType)));
             il.Append(il.Create(OpCodes.Stloc_0));
 
-            foreach (var kvp in properties)
+            foreach (var prop in properties)
             {
-                var property = kvp.Key;
-                var field = kvp.Value;
+                var property = prop.Property;
+                var field = prop.Field;
 
                 if (property.SetMethod != null)
                 {
-                    if (!IsNullable(property))
+                    Instruction updatePlaceholder = null;
+                    Instruction managePlaceholder = null;
+                    if (IsRealmObjectDescendant(property))
                     {
                         il.Append(il.Create(OpCodes.Ldloc_0));
                         il.Append(il.Create(OpCodes.Ldfld, field));
 
+                        managePlaceholder = il.Create(OpCodes.Nop);
+                        il.Append(managePlaceholder);
+
+                        il.Append(il.Create(OpCodes.Ldloc_0));
+                        il.Append(il.Create(OpCodes.Call, _realmObjectRealmGetter));
+                        il.Append(il.Create(OpCodes.Ldloc_0));
+                        il.Append(il.Create(OpCodes.Ldfld, field));
+                        il.Append(il.Create(OpCodes.Ldarg_2));
+                        il.Append(il.Create(OpCodes.Call, new GenericInstanceMethod(_realmManageGenericReference) { GenericArguments = { field.FieldType } }));
```

Comment:
I hadn't introduced the non-generic one at that point :) I think it's fine to leave it like that, but I'll need to ensure that the correct `MethodReference` is obtained.